Source/Csla.AspNetCore/Blazor/ActiveCircuitHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿//-----------------------------------------------------------------------
```

Comment:
Two CircuitManagers cannot be registered...so I'm wondering what happens when I register my own custom circuit handler, won't that step on this guys' toes?

Reply:
My understanding - or maybe assumption - was that you could have as many circuit handlers as you want. I think it probably uses the same IEnumerable<T> feature of DI as we use in Csla 6 for interceptors. Have you found some documentation that says otherwise?

Reply:
In the demo I'm working on (haven't uploaded it yet), I have a "scoped" circuit handler that I'm making heavy use of.  Yesterday as I was adding more features, I was wondering "can I have a singleton circuit handler too?".  So I made one and registered it in Program.cs.  
All of a sudden all my "scoped" ones were null.  So I just assumed DI wouldn't allow two.  But maybe it had to do with the different scopes?

Reply:
No, a singleton circuit handler wouldn't work - or it would do something extremely strange, at the very least. At a guess there would be about a 15% chance that it would create a black hole that would engulf the Earth and grow to the size of the universe. Something like that. I wouldn't risk it ;-)

The concept of a circuit handler only makes sense within a circuit - which represents a scope. The circuit manager (or another, related class) creates a scope for the circuit as it is created. From there on, it's scopes all the way down.

Reply:
LOL - Superman 3

I agree.  I was surprised when I saw [this Microsoft article](https://github.com/MarimerLLC/csla/discussions/2687#discussioncomment-2051923) and decided to try it for myself...wondering "how the heck would this work?!"  Notice they register it as a singleton.  Perhaps the documentation person just made a mistake??

Reply:
Hmm, yeah, interesting. You linked back to us, not the article, but I think I see the one you mean.

Here is CircuitHost, it accepts an array of circuit handlers into its constructor and iterates through all of them in the appropriate places, so there should be no reason why there can't be lots.

https://github.com/dotnet/aspnetcore/blob/main/src/Components/Server/src/Circuits/CircuitHost.cs

I was expecting to see `IEnumerable<CircuitHandler>` but I guess an array does the same thing. I know they are somewhat interchangeable in places.

Reply:
Sorry...fixed the link above.

Reply:
Thanks. Yeah, it was the one I found.

It makes sense that you could use a handler across circuits I suppose, as long as you don't expect it to do anything specific to any single one.

I wonder if you might have got caught out by captive dependencies somehow in what you were doing. It's just a guess, but it might have explained what you were seeing. https://blog.ploeh.dk/2014/06/02/captive-dependency/

Anyway, it seems .NET allows for there being multiple, which is what we were concerned about.

Reply:
Thank you both for the deep thinking here.

Just so I'm clear: what you discovered is that we can register this circuit handler, and people can register other ones too, and they'll be honored (if they are scoped the same)?

(I wonder if the confusion is that in Blazor wasm Singleton and Scoped have the same effect in most cases, but in SSB they always have different effects? Maybe the sample author was confused about this?)